### PR TITLE
Add Custom questions demos

### PR DESCRIPTION
**Pull request by gonzalozawa**
@mcgoughn this is ready for review. Please let me know your comments.

### CODE REVIEW

**Comment by gonzalozawa** [comment]
@denishoctor, Neil asked me to add custom questions demos to the demo site. I have added the following:
- Custom Short Text
- Custom Percentage Bar
- Custom Mathcore
- Custom Box Whisker
- Custom Implementation
They are ready for review in this branch (custom-questions). Could you please have a look and let me know if you are happy with them so I can merge them to master?
Please let me know your comments.
Cheers.
Gonzalo

File: www/casestudies/items/customquestions/custom.php [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    use LearnositySdk\Request\Init;
+    use LearnositySdk\Utils\Uuid;
+
+    $session_id = Uuid::generate();
+
+    $security = [
+    'user_id'      => $studentid,
+    'domain'       => $_SERVER['SERVER_NAME'],
+    'consumer_key' => $consumer_key,
+    'timestamp'    => gmdate('Ymd-Hi')
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom.php]: Fine to use this, but the SDK generates it for you so I usually omit.

File: www/casestudies/items/customquestions/custom.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    ];
+
+    $init = new Init('questions', $security, $consumer_secret, [
+        'id'                   => 'custom-shorttext',
+        'name'                 => 'Custom Short Text',
+        'course_id'            => $courseid,
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom.php]: Just so you know, `course_id` is no longer required so is good to omit.

File: www/casestudies/items/customquestions/custom.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            <button class="btn btn-primary pull-right" id="render_custom_question">Render JSON</button>
+        </div>
+        <div class="col-md-6">
+            <div class="custom-score"><strong>Score: </strong> <span id="question_score">0</span> / <span id="question_max_score">0</span></div>
+            <button class="btn btn-primary pull-right" id="validate_question">Check Answer</button>
+        </div>
+    </div>
+</div>    
+
+<script src="//questions.learnosity.com"></script>
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom.php]: We use variables for API URLs now, see any of the other demos. Means they work across all environments (vg, staging and prod).

File: www/casestudies/items/customquestions/custom_box_whisker.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    use LearnositySdk\Request\Init;
+    use LearnositySdk\Utils\Uuid;
+
+    $session_id = Uuid::generate();
+
+    $security = [
+    'user_id'      => $studentid,
+    'domain'       => $_SERVER['SERVER_NAME'],
+    'consumer_key' => $consumer_key,
+    'timestamp'    => gmdate('Ymd-Hi')
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom_box_whisker.php]: As above with the timestamp property and indenting.

File: www/casestudies/items/customquestions/custom_box_whisker.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            "params_box2_color"    :       "#999999",
+            "valid_range_1"        :       2, 
+            "valid_range_2"        :       11, 
+            "valid_quartile_1"     :       4, 
+            "valid_median"         :       6, 
+            "valid_quartile_3"     :       8.5,
+            "score": 1
+        }]
+    }';
+
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom_box_whisker.php]: Extraneous line break.

File: www/casestudies/items/customquestions/custom_box_whisker.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    <div class="row">
+        <div class="question-container">
+            <span class="learnosity-response question-custom-box-whisker-response-1"></span>
+            <div class="custom-score"><strong>Score: </strong> <span id="question_score">0</span> / <span id="question_max_score">0</span></div>            
+            <button class="btn btn-primary pull-right" id="validate_question">Check Answer</button>
+        </div>
+    </div>
+</div>    
+
+<script src="//questions.learnosity.com"></script>
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom_box_whisker.php]: As above with variable to manage cross-environment values.

File: www/casestudies/items/customquestions/custom_mathcore.php [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    use LearnositySdk\Request\Init;
+    use LearnositySdk\Utils\Uuid;
+
+    $session_id = Uuid::generate();
+
+    $security = [
+    'user_id'      => $studentid,
+    'domain'       => $_SERVER['SERVER_NAME'],
+    'consumer_key' => $consumer_key,
+    'timestamp'    => gmdate('Ymd-Hi')
```
Comment (michaelsharman) [inline www/casestudies/items/customquestions/custom_mathcore.php]: As above with timestamp and indenting.

File: src/includes/nav.php [hunk 8]
```diff
@@ -21,7 +21,8 @@
             'casestudies/items/printing' => 'Printing Items',
             'casestudies/items/gallery'  => 'Gallery Style UI',
             'casestudies/items/xapi'     => 'xAPI Events',
-            'casestudies/items/endtoend' => 'End to End Demo'
+            'casestudies/items/endtoend' => 'End to End Demo',
+            'casestudies/items/customquestions' => 'Custom Questions',
         )
```
Comment (michaelsharman) [inline src/includes/nav.php]: FYI I merged a PR that removed the `items` string from the path.

**Comment by gonzalozawa** [comment]
@michaelsharman thanks for the comments. I have updated the branch and PR. Could you please let me know if you are happy with it? Cheers :)

**Comment by michaelsharman** [comment]
A couple of small things, but let's not hold it back any further. I didn't functionally review, just browsed through the code.

**Pull request merged**